p2p/bootstrap.go

```diff
@@ -67,20 +67,21 @@ func (m *p2pModule) bootstrap() error {
 
 		currentHeightProvider := rpcCHP.NewRPCCurrentHeightProvider(rpcCHP.WithCustomRPCURL(bootstrapNode))
 
-		addrBook, err = addressBookProvider.GetStakedAddrBookAtHeight(currentHeightProvider.CurrentHeight())
+		pstore, err = addressBookProvider.GetStakedAddrBookAtHeight(currentHeightProvider.CurrentHeight())
 		if err != nil {
 			m.logger.Warn().Err(err).Str("endpoint", bootstrapNode).Msg("Error getting address book from bootstrap node")
 			continue
 		}
 	}
 
-	if len(addrBook) == 0 {
+	if pstore.Size() == 0 {
 		return fmt.Errorf("bootstrap failed")
 	}
 
-	for _, peer := range addrBook {
-		m.logger.Debug().Str("address", peer.Address.String()).Msg("Adding peer to addrBook")
-		if err := m.network.AddPeerToAddrBook(peer); err != nil {
+	for _, peer := range pstore.GetAllPeers() {
+		m.logger.Debug().Str("address", peer.GetAddress().String()).Msg("Adding peer to pstore")
+		// TECHDEBT: either remove the returned error from the interface OR log the error here.
+		if err := m.network.AddPeer(peer); err != nil {
```

Comment:
Thoughts on logging a fatal error? Bootstrapping shouldn't fail

Reply:
I think that comment was not ready for ready for review - hehe. Re-reading it, I'm not sure exactly what I was thinking. I've removed it and logged a non-fatal error. I agree that it "shouldn't" fail but there are several opportunities for an error to be produced downstream of `Network#AddPeer` now that libp2p is involved. I think we will reduce these as we get our integration tighter going forward. Regardless, I don't think an error at this point should terminate the program. So long as adding every bootstrap peer doesn't fail, I think we should be fine to continue. I also repurposed the `pstore.Size == 0` check and moved it to the end:

```
	if m.network.GetPeerstore().Size() == 0 {
		return fmt.Errorf("bootstrap failed")
	}
```